include/mbedtls/x509_crt.h

```diff
@@ -156,13 +156,17 @@ mbedtls_x509_subject_alternative_name;
  * Security profile for certificate verification.
  *
  * All lists are bitfields, built by ORing flags from MBEDTLS_X509_ID_FLAG().
+ *
+ * The fields of this structure are part of the public APi and can be
+ * manipulated directly by applications. Future versions of the library may
+ * add extra fields.
```

Comment:
Suggestion:
```suggestion
 * add extra fields or reorder fields.
```

Reply:
Let's start with saying we may reorder fields, and we can lift this self-imposed restriction later.

Reply:
I think we should document whether applications are are expected to always start from a built-in profile and modify it, or if they can also start with the all-0 profile - that is, is and when we add fields, do we commit to the value 0 for the new field to always mean the same behavior as before that field was added?

Reply:
I think given that we define mbedtls_x509_crt_profile_default it seems not particularly useful to support starting from the all-0 profile? +1 for documenting this expectation though.

Reply:
> it seems not particularly useful to support starting from the all-0 profile

It would make sense to start from an all-is-forbidden rather than from a sensible-default profile if you want to have precise control over what to allow. Unfortunately, all-zero isn't the ideal all-is-forbidden profile: it does mean all is forbidden, but if you start allowing RSA by adding it to `allowed_pks`, you'll allow RSA keys of any size — a better all-is-forbidden profile would have `rsa_min_bitlen == impossibly_high`.

Reply:
I pushed something which I think clarifies expectations and supports the use case Gilles was mentioning.